arbutil/espresso_utils.go

```diff
@@ -38,6 +40,11 @@ func BuildRawHotShotPayload(
 			break
 		}
 
+		if len(msgBytes) >= MAX_L2_MESSAGE_SIZE {
+			log.Warn("message too large", "pos", p, "size", len(msgBytes))
+			break
+		}
+
```

Comment:
The HotShot payload actually includes both the message and its index.

In the Espresso streamer, it always looks for the message with the next expected index. So, if any message index is skipped, it breaks the entire workflow.

It would be better if we could avoid receiving excessively large messages.

BTW, if we break at this point, the next caller will also pick up the same index. Since it's the earliest available, it will halt further processing.

Reply:
Thank you for the very clear explanations @ImJeremyHe, that helps a lot

Reply:
New check added at this commit: [e1d1e01](https://github.com/EspressoSystems/nitro-espresso-integration/pull/617/commits/e1d1e0181f5fd68daf97562a803b06bdece12bf5)